auraed/src/runtime/cells/executable.rs

```diff
@@ -135,11 +137,39 @@ impl Executable {
     }
 }
 
-fn aurae_process_pre_exec(executable_name: &ExecutableName) -> io::Result<()> {
-    info!("CellService: aurae_process_pre_exec(): {executable_name}");
+/// Common functionality within the context of the new executable
+fn pre_exec(
+    executable_name: &ExecutableName,
+    spec: ValidatedCell,
+) -> io::Result<()> {
+    info!("CellService: pre_exec(): {executable_name}");
     // Here we are executing as the new spawned pid.
     // This is a place where we can "hook" into all processes
     // started with Aurae in the future. Similar to kprobe/uprobe
     // in Linux or LD_PRELOAD in libc.
+
+    // In the event we are not sharing the mount namespace or the pid namespace
+    // with the host, we will manually mount /proc
+    if !spec.ns_share_pid && !spec.ns_share_mount {
+        info!("CellService: pre_exec(): mounting procfs /proc");
+        let supported = SupportedFilesystems::new().unwrap();
+        info!(
+            "CellService: pre_exec(): Supported filesystems: {:?}",
+            supported
+        );
+        let mount_result =
+            Mount::builder().fstype("proc").mount("proc", "/proc");
+        match mount_result {
+            Ok(mount) => {
+                // TODO We need to figure out how to manage this flag AFTER the command has been executed
+                // TODO Either on kill or stop we need to call this
```

Comment:
so we need a post_exec callback?

Reply:
I think so -- The more I think about it the more I think we are going to be pre and post hooks as a service... unfortunately 

There is going to be a lot of logic that will end up in the pre and post hooks... a lot...